pkg/resource/config/value.go

```diff
@@ -42,47 +53,126 @@ func (c Value) Value(decrypter Decrypter) (string, error) {
 	if decrypter == nil {
 		return "", errors.New("non-nil decrypter required for secret")
 	}
+	if c.object && decrypter != NopDecrypter {
+		var obj interface{}
+		if err := json.Unmarshal([]byte(c.value), &obj); err != nil {
+			return "", err
+		}
+		decryptedObj, err := decryptObject(obj, decrypter)
+		if err != nil {
+			return "", err
+		}
+		json, err := json.Marshal(decryptedObj)
+		if err != nil {
+			return "", err
+		}
+		return string(json), nil
+	}
 
 	return decrypter.DecryptValue(c.value)
 }
 
+func (c Value) SecureValues(decrypter Decrypter) ([]string, error) {
+	d := NewTrackingDecrypter(decrypter)
```

Comment:
Why is the tracking Decrypter needed here? Can't we just keep keep appending to a slice of values we have decrypted already?

If we are only using the new `TrackingDecrypter` in one place, I'm not sure it warrants introducing another type of decrypted when it isn't _doing_ much for us.

Reply:
I need a way to get the plaintext values of all secure values in an object, so we can add those to the log filter (to blind such secret values from log output).

I'd already written the code that walks the object to decrypt such secure values:

https://github.com/pulumi/pulumi/blob/88491278472ffae367e61768b9a4c304972f0f10/pkg/resource/config/value.go#L243-L275

I created `TrackingDecrypter` to pass to `decryptObject` as a simple way of keeping track of all the values in the object that were decrypted.

An alternative would be to change the implementation of `decryptObject` to also build-up an array of decrypted values as it is recursively walking the object, but that'll make the implementation of `decryptObject` more complex. Or write a new similar function that walks the object in the same way, but builds-up an array rather than returning a new object, but that'd duplicate some logic.